Automattic/woocommerce-payments
# Feature/1816 empty state deposits screen

Fixes #1816 #1917 

This PR adds empty state Deposits and Transaction screens for the `wcpay_empty_state_preview_mode` experiment.

#### Changes proposed in this Pull Request

1. Created a new directory `empty-state-table` and added any empty state related files and components to minimize changes to the existing codes.
2. Added both Deposits and Transaction screens in this PR since they both share the exact same design, except for the list image.


#### Testing instructions


1. Turn on your A8C proxy.
2. Find `wcpay_empty_state_preview_mode` experiment in Abcus. Add both `control` and `treatment` variations to your bookmark.
3. Open `includes/admin/class-wc-payments-admin.php` file. Find this [line](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/admin/class-wc-payments-admin.php#L88) and add `$should_render_full_menu=true`. This forces WC Payment menu to show up. This change is for testing purposes only and temporary until we have https://github.com/Automattic/woocommerce-payments/issues/1815 ready
4. Login to your WC Admin and assign yourself to `treatment` group by clicking the bookmark.

**Testing Deposits**

1. Navigate to Payments -> Deposits. You should see the preview mode (screenshot #1)
2. Turn on tracking and click the overview area. You should see `wcadmin_wcpay_deposits_summary_empty_state_click` event.
3. Click "Finish setup". You should see `wcadmin_wcpay_connect_account_clicked` event and it should redirects you to WC Payments KYC.

**Testing Transactions**

1. Navigate to Payments -> Deposits. You should see the preview mode (screenshot #2)
2. Click "Finish setup". You should see `wcadmin_wcpay_connect_account_clicked` event and it should redirects you to WC Payments KYC.

**Testing control**

1. Assign yourself to `control` group
2. Navigate to Payments -> Deposits and confirm that you're seeing the default experience.
3. Navigate to Payments -> Transaction and confirm that you're seeing the default experience.


#### Screenshots

![Screen Shot 2021-06-12 at 3 04 58 PM](https://user-images.githubusercontent.com/4723145/121790044-1bf98a80-cb90-11eb-95c2-dd5d4cf35731.jpg)
![Screen Shot 2021-06-12 at 3 05 03 PM](https://user-images.githubusercontent.com/4723145/121790045-1ef47b00-cb90-11eb-8edb-554a9de177e3.jpg)
